### PR TITLE
Add autocomplete users to project edit scoped by membership

### PR DESCRIPTION
**Pull request by todtb**
#2670 
~Do Not Merge~

~I've opened this as a draft as we will need the endpoint merged here: https://github.com/inaturalist/iNaturalistAPI/pull/195~

Also this adds text to the Project edit page which will need i18n.

Other than that it's working great with the ES scoped to users already in the project on the API end and the controller rejecting invalid options to be safe.

Note: I've left the ability to add admins on the Project new page as there are no members yet and this could save some time and page loads.

### CODE REVIEW

**Comment by todtb** [comment]
@kueda @pleary Updated to match changes to https://github.com/inaturalist/iNaturalistAPI/pull/195

I suppose the API prerequisite is no longer applicable, but I still think it makes sense to get to that first.

**Comment by todtb** [comment]
https://github.com/inaturalist/iNaturalistAPI/pull/198 should allow this to go forward until the index is added, unless the 1000 member cap would prevent doing so.

File: app/assets/javascripts/i18n/translations.js [hunk 1]
```diff
@@ -17133,6 +17133,7 @@ I18n.translations["en"] = {
     "projects": {
       "collection": "Collection",
       "edit": {
+        "admins_must_be_existing_members": "Admins must be existing members of the project.",
```
Comment (kueda) [inline app/assets/javascripts/i18n/translations.js:17136]: This needs to be specified in https://github.com/inaturalist/inaturalist/blob/master/config/locales/en.yml#L5675. This js file will get overwritten with every deploy.

File: app/models/project.rb [hunk 2]
```diff
@@ -1038,16 +1038,16 @@ def flagged_with( flag, options = {} )
...
-        if pu = project_users.where( user_id: admin_attr["user_id"] ).first
-          pu.update_attributes( role: nil )
-        end
+      new_role = admin_attr["_destroy"] == "true" ? nil : "manager" 
+      if new
+        project_users.find_or_create_by( user_id: admin_attr["user_id"] ).update_attributes( role: new_role )
       else
-        pu = project_users.find_or_create_by( user_id: admin_attr["user_id"] )
-        pu.update_attributes( role: "manager" )
+        project_users.find_by( user_id: admin_attr["user_id"] )&.update_attributes( role: new_role )
```
Comment (kueda) [inline app/models/project.rb:1050]: Neat, I didn't know about the `&.` operator.

File: app/assets/javascripts/i18n/translations.js (hunk 1 above)
Comment (todtb) [reply app/assets/javascripts/i18n/translations.js:17136]: Ahh, I see `generate_translations_js` rake task. Thanks for the info. I guess changing this output is irrelevant and unnecessary.

**Comment by kueda** [comment]
Looking good. Thanks again!

**Pull request merged**